sparkctl/cmd/create.go

```diff
@@ -329,15 +331,21 @@ func buildHadoopConfigMap(appName string, hadoopConfDir string) (*apiv1.ConfigMa
 		if err != nil {
 			return nil, err
 		}
-		hadoopConfigFiles[file.Name()] = string(content)
+
+		if utf8.Valid(content) {
```

Comment:
What if the file encoding is not utf-8?

Reply:
It gets stored as new type: binaryData. For legacy configMap, the files with non-utf8 were anyways corrupted, since they were converted to string. So in the cluster you could see the data, but when you try using it, you get errors about bad format or corruption. I guess, with binaryData flag for legacy configMap, it will be just ignored (instead of being corrupted) - probably worth testing

Reply:
Copied from 1.10 change logs: `ConfigMap objects now support binary data via a new binaryData field. When using kubectl create configmap --from-file, files containing non-UTF8 data will be placed in this new field in order to preserve the non-UTF8 data.`. So this looks like the right approach. I think it worths testing this in a 1.9 cluster.